near-sdk/src/store/lookup_map/mod.rs

```diff
@@ -0,0 +1,533 @@
+        entry.value().as_ref()
+    }
+
+    fn get_mut_inner<Q: ?Sized>(&mut self, k: &Q) -> &mut CacheEntry<V>
+    where
+        K: Borrow<Q>,
+        Q: BorshSerialize + ToOwned<Owned = K>,
+    {
+        let prefix = &self.prefix;
+        //* ToOwned bound, which forces a clone, is required to be able to keep the key in the cache
+        let entry = self.cache.get_mut(k.to_owned());
+        entry.get_or_init(|| CacheEntry::new_cached(Self::load_element(prefix, k)));
+        let entry = entry.get_mut().unwrap_or_else(|| unreachable!());
+        entry
+    }
+
+    /// Returns a mutable reference to the value corresponding to the key.
+    ///
+    /// The key may be any borrowed form of the map's key type, but
+    /// [`BorshSerialize`] and [`ToOwned<Owned = K>`](ToOwned)on the borrowed form *must* match those for
+    /// the key type.
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut V>
+    where
+        K: Borrow<Q>,
+        Q: BorshSerialize + ToOwned<Owned = K>,
+    {
+        self.get_mut_inner(k).value_mut().as_mut()
+    }
+
+    /// Inserts a key-value pair into the map.
+    ///
+    /// If the map did not have this key present, [`None`] is returned.
+    ///
+    /// If the map did have this key present, the value is updated, and the old
+    /// value is returned. The key is not updated, though; this matters for
+    /// types that can be `==` without being identical.
+    pub fn insert(&mut self, k: K, v: V) -> Option<V>
+    where
+        K: Clone,
+    {
+        self.get_mut_inner(&k).replace(Some(v))
+    }
+
+    /// Returns `true` if the map contains a value for the specified key.
+    ///
+    /// The key may be any borrowed form of the map's key type, but
+    /// [`BorshSerialize`] and [`ToOwned<Owned = K>`](ToOwned)on the borrowed form *must* match those for
+    /// the key type.
+    pub fn contains_key<Q: ?Sized>(&self, k: &Q) -> bool
+    where
+        K: Borrow<Q>,
+        Q: BorshSerialize + ToOwned<Owned = K> + Ord,
+    {
+        // Check cache before checking storage
+        if self
+            .cache
+            .with_value(&k, |v| v.get().and_then(|s| s.value().as_ref()).is_some())
+            .unwrap_or(false)
+        {
+            return true;
+        }
+        let storage_key = Self::lookup_key(&self.prefix, k);
+        let contains = env::storage_has_key(&storage_key);
+
+        if !contains {
+            // If value not in cache and not in storage, can set a cached `None`
+            let _ = self.cache.get(k.to_owned()).set(CacheEntry::new_cached(None));
+        }
+        contains
+    }
+
+    /// Removes a key from the map, returning the value at the key if the key
+    /// was previously in the map.
+    ///
+    /// The key may be any borrowed form of the map's key type, but
+    /// [`BorshSerialize`] and [`ToOwned<Owned = K>`](ToOwned)on the borrowed form *must* match those for
```

Comment:
idk how this looks like when rendered, but might want a space here?
```suggestion
    /// [`BorshSerialize`] and [`ToOwned<Owned = K>`](ToOwned) on the borrowed form *must* match those for
```